src/libraries/System.Private.CoreLib/src/System/Double.cs

```diff
@@ -167,19 +167,11 @@ public static unsafe bool IsInfinity(double d)
             return (bits & 0x7FFFFFFFFFFFFFFF) == 0x7FF0000000000000;
         }
 
-        /// <summary>Determines whether the specified value is NaN.</summary>
+        /// <summary>Determines whether the specified value is NaN. A NaN will never equal itself so this is an easy and the most efficient way to check for NaN.</summary>
         [NonVersionable]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static unsafe bool IsNaN(double d)
-        {
-            // A NaN will never equal itself so this is an
-            // easy and efficient way to check for NaN.
-
-            #pragma warning disable CS1718
-            return d != d;
-            #pragma warning restore CS1718
-        }
-
+        public static unsafe bool IsNaN(double d) => false;
```

Comment:
NaN won't equal itself and so `d != d` returns `true` for `NaN` and `false` otherwise.

Always returning `false` would violate the check and be an invalid change.